web/cobrands/fixmystreet/assets.js

```diff
@@ -445,11 +442,79 @@ fixmystreet.assets = {
             fixmystreet.map.addControl(fixmystreet.assets.controls[i]);
             fixmystreet.assets.controls[i].activate();
         }
+    },
+
+    select_usrn: function(lonlat) {
+        var usrn_providers = fixmystreet.map.getLayersBy('fixmystreet', {
+            test: function(options) {
+                return options && options.usrn_field;
+            }
+        });
+        if (usrn_providers.length) {
+            var usrn_layer = usrn_providers[0];
+            var point = new OpenLayers.Geometry.Point(lonlat.lon, lonlat.lat);
+            var feature = usrn_layer.getFeatureAtPoint(point);
+            if (feature == null) {
+                // The click wasn't directly over a road, try and find one
+                // nearby
+                feature = usrn_layer.getNearestFeature(point, 10);
+            }
+            if (feature !== null) {
+                selected_usrn = feature.attributes[usrn_layer.fixmystreet.usrn_field];
+            } else {
+                selected_usrn = null;
+            }
+            fixmystreet.assets.update_usrn_field();
+        }
+    },
+
+    update_usrn_field: function() {
+        $("input[name=usrn]").val(selected_usrn);
     }
 };
 
 $(function() {
     fixmystreet.assets.init();
 });
 
+OpenLayers.Layer.Vector.prototype.getFeatureAtPoint = function(point) {
+    for (var i = 0; i < this.features.length; i++) {
+        var feature = this.features[i];
+        if (feature.geometry.containsPoint && feature.geometry.containsPoint(point)) {
```

Comment:
Do we know all features will have geometries?

Reply:
Hah, no, they won't. I just discovered this the hard way. Will fix!